src/components/screens/InitialScreen.tsx

```diff
@@ -27,37 +25,38 @@ const InitialScreen = ({ mint, setIsWalletModalOpen, isWalletModalOpen }: Props)
     setForm,
     setErrors,
     selectedSwapRoute,
-    mode,
+    formProps: {
+      initialOutputMint,
+      fixedOutputMint,
+    },
     jupiter: { loading },
   } = useSwapContext();
   const { setScreen } = useScreenState();
 
   const walletPublicKey = useMemo(() => wallet?.adapter.publicKey?.toString(), [wallet?.adapter.publicKey]);
 
-  useEffect(() => {
-    setForm((prev) => ({ ...prev, toMint: mint }));
-  }, [mint]);
-
-  // TODO: Dedupe the balance
   const balance = useMemo(() => {
     return form.fromMint ? accounts[form.fromMint]?.balance || 0 : 0;
   }, [walletPublicKey, accounts, form.fromMint]);
 
-  const isDisabled = useMemo(() => {
+  const [isDisabled, setIsDisabled] = useState(false);
+  useEffect(() => {
```

Comment:
not sure why the change to state

Reply:
had some console errors during mounting
did not understand the full extent of it, but might be related to useMemo being called before render, and setErrors called too early and the component wasn’t mounted properly yet